### ISSUE TITLE
gcc 10 incompatibility still present in IGAMX2D call in pstrsen.f and pstrord.f

### ISSUE THREAD

**Issue by traversaro** [Reference-ScaLAPACK/scalapack#55]
First of all, thanks for the work on scalapack and sorry if the issue is not relevant. 

I am trying to compile scalapack 2.2.0 with gcc 10, and I am getting error of type:
~~~
[ 45%] Building Fortran object CMakeFiles/scalapack.dir/SRC/pstrsen.f.o
/home/conda/feedstock_root/build_artifacts/scalapack_1646690481794/work/SRC/pstrsen.f:608:48:

  529 |      $           CALL IGAMX2D( ICTXT, 'All', TOP, 1, 1, MMAX, 1, -1,
      |                                                      2
......
 3253 |             CALL IGAMX2D( ICTXT, 'All', TOP, 1, 1, IERR, 1, -1,
      |                                                   1
Error: Rank mismatch between actual argument at (1) and actual argument at (2) (rank-1 and scalar)
/home/conda/feedstock_root/build_artifacts/scalapack_1646690481794/work/SRC/pstrord.f:3264:54:

  490 |      $         CALL IGAMX2D( ICTXT, 'All', TOP, 1, 1, MMAX, 1, -1,
      |                                                      2
......
 3264 |                CALL IGAMX2D( ICTXT, 'All', TOP, 1, 1, INFO, 1, -1,
      |                                                      1
Error: Rank mismatch between actual argument at (1) and actual argument at (2) (rank-1 and scalar)
make[2]: *** [CMakeFiles/scalapack.dir/build.make:12812: CMakeFiles/scalapack.dir/SRC/pstrord.f.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1234: CMakeFiles/scalapack.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
Traceback (most recent call last):
~~~

In theory all the problems of this kind should have been fixed by https://github.com/Reference-ScaLAPACK/scalapack/pull/23, but on that specific file we are still getting them. Notice that the same error was also reported by the Homebrew CI, see https://github.com/Homebrew/homebrew-core/pull/94420/files#diff-e45b9bfcc61ceafda0d6200c3c06dec48c0da5a6259d8346659d3fe822b0ae0bL35 .

**Comment by weslleyspereira** [comment]
Hi @traversaro. We merged #51. Should I close issue #55 ?

**Comment by traversaro** [comment]
Ok for me, thanks!
